### PR TITLE
feat(templating): adds encodeBase64 handlebars helper

### PR DESCRIPTION
**Pull request by moltar**
## Changes

Adds a `encodeBase64` handlebars helper that encodes a string to Base64.

## Context

When using template variables with strings that may contain special characters, it makes it more
difficult to use with commands.

Consider this example:

```json
"postUpgradeTasks": [
  "echo \"{{body}}\" > file.md"
]
```

The value of the `body` variable contains the full changelog, which may contain _anything_.

Renovate will replace the variable, _before_ the command runs, and thus it will result in a command:

```sh
echo "This is a changelog" > file.md
```

But if the `body` variable contains a double quote, the command will fail:

```sh
echo "This is "a" changelog" > file.md
```

By encoding the values in Base64, we can avoid this issue:

```json
"postUpgradeTasks": [
  "echo \"{{encodeBase64 body}}\" | base64 -d > file.md"
]
```

Related discussion: <https://github.com/renovatebot/renovate/discussions/26175>

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

**Review by secustor — DISMISSED** [review]

File: lib/util/template/index.ts [hunk 1]
```diff
@@ -8,6 +8,10 @@ import { regEx } from '../regex';
 handlebars.registerHelper('encodeURIComponent', encodeURIComponent);
 handlebars.registerHelper('decodeURIComponent', decodeURIComponent);
 
+handlebars.registerHelper('encodeBase64', (str: string) =>
+  Buffer.from(str).toString('base64'),
```
Comment (viceice) [inline lib/util/template/index.ts:12]:
```suggestion
  Buffer.from(str ?? '').toString('base64'),
```

should we protect against null or undefined?

Comment (rarkins) [reply lib/util/template/index.ts]: This has probably impacted code coverage now :)

Comment (moltar) [reply lib/util/template/index.ts]: I thought about this, but then I didn't see other helpers do that, so I figured I'd follow the same pattern for now.

**Review by viceice — DISMISSED** [review]

**Comment by moltar** [comment]
Added extra tests for nullish coalescing to fix coverage.

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**